contracts/Invest.sol

```diff
@@ -133,6 +117,18 @@ contract Invest is PoolsData {
         RegisterInvest(_PoolId, Tokens);
     }
 
+    function TokenAllocate(uint256 _PoolId, uint256 _ThisInvestor, uint256 _Tokens) internal {
+        if (isPoolLocked(_PoolId)) {
+            Investors[_ThisInvestor].TokensOwn = SafeMath.add(
+                Investors[_ThisInvestor].TokensOwn,
+                _Tokens
+            );
+        } else {
+            // not locked, will transfer the tokens
+            TransferToken(pools[_PoolId].BaseData.Token, msg.sender, _Tokens);
```

Comment:
I think it's better not to use msg.sender in the internal function.
we got the address in the _thisInvestor